android/kotlin/v2/AmpliApp/app/src/main/java/com/amplitude/ampli/Ampli.kt

```diff
@@ -0,0 +1,822 @@
+//
+// Ampli - A strong typed wrapper for your Analytics
+//
+// This file is generated by Amplitude.
+// To update run 'ampli pull kotlin-ampli-v2'
+//
+// Required dependencies: com.amplitude:analytics-android:0.1.0-beta.6
+// Tracking Plan Version: 0
+// Build: 1.0.0
+// Runtime: android:kotlin-ampli-v2
+//
+// [View Tracking Plan](https://data.amplitude.com/test-codegen/Test%20Codegen/events/main/latest)
+//
+// [Full Setup Instructions](https://data.amplitude.com/test-codegen/Test%20Codegen/implementation/kotlin-ampli-v2)
+//
+
+package com.amplitude.ampli
+
+import com.amplitude.android.Amplitude
+import com.amplitude.android.Configuration
+import com.amplitude.android.events.BaseEvent
+import com.amplitude.android.events.EventOptions
+import com.amplitude.android.events.Plan
+
+enum class EventType(val value: String) {
+    Identify("\$identify"),
+    GroupIdentify("\$groupidentify"),
+}
+
+class LoadOptions(
+    val environment: Ampli.Environment? = null,
+    val disabled: Boolean? = null,
+    val client: LoadClientOptions? = null
+)
+
+class LoadClientOptions(
+    val apiKey: String? = null,
+    val instance: Amplitude? = null,
+    val plan: Plan? = null
+)
+
+class Identify private constructor() : BaseEvent() {
```

Comment:
Don't know if I understand your question right. Let me know if not. 
1. I don't think we can prevent class duplication. But when they run `ampli pull`, the auto-generated file will have conflicts, so they must deal with the duplicate name, they might change their event name.  I think for all languages, if one class is a reserved class, then we cannot re-define the class with the same name. 
2. The IdentifyEvent is not been exposed in the previous android-kotlin version. So I used the BaseEvent instead of IdentifyEvent.  Since the user may define some user properties in `Data` ui, we need to auto-generate those user properties. This class just extends the BaseEvent and overwrites the user properties. 
In Ampli we only allow one operation for user properties which is `set`.  So in order to make it works, I think we can either call an Amplitude Identify class and using `set` to set the user properties and call identify or we can set the user properties to a key value map and call the identify the call. I add a new function signature for identify call in kotlin. It's allowed to pass a map. 

In [ampli-kotlin-v1](https://github.com/amplitude/ampli-examples/blob/main/android/kotlin/AmpliApp/app/src/main/java/com/amplitude/ampli/Ampli.kt#L63), we didn't make an identify call, where did you say we make an identify call instead?